source/js/RecipeExpand.js

```diff
@@ -0,0 +1,496 @@
+  
+      // Set Ratings
+      const ratingVal = searchForKey(data, 'ratingValue');
+      let ratingTotal = searchForKey(data, 'ratingCount');
+      const rating = this.shadowRoot.querySelector('.rating--wrapper');
+      const numStars = Math.round(ratingVal);
+      if (ratingVal) {
+        rating.innerHTML = `
+        <img src="assets/images/icons/${numStars}-star.svg" alt="${numStars} stars">
+        <span>${ratingVal}</span>
+        from
+        `;
+        if (!ratingTotal) {
+          ratingTotal = 'some';
+        }
+        rating.innerHTML += `<span class="rating-total">${ratingTotal} votes</span>`;
+      } else {
+        rating.innerHTML = `
+          <span>No Reviews</span>
+        `;
+      }
+  
+      // Set Ingredients
+      const ingredients = getIngredients(data);
+      ingredients.forEach(ingredient => {
+        const listItem = document.createElement('li');
+        listItem.innerHTML = ingredient;
+        this.shadowRoot.querySelector('.section--ingredients > ul').append(listItem);
+      });
+  
+      // Set Instructions
+      const instructions = getInstructions(data);
+      instructions.forEach(instruction => {
+        const listItem = document.createElement('li');
+        listItem.innerHTML = instruction;
+        this.shadowRoot.querySelector('.section--instructions > ol').append(listItem);
+      });
+    }
+  
+    /**
+     * Returns the object of the currect recipe being used.
+     */
+    get data() {
+      return this.json;
+    }
+  }
+  
+  /**
+   * Recursively search for a key nested somewhere inside an object
+   * @param {Object} object the object with which you'd like to search
+   * @param {String} key the key that you are looking for in the object
+   * @returns {*} the value of the found key
+   */
+  function searchForKey(object, key) {
+    var value;
+    Object.keys(object).some(function (k) {
+      if (k === key) {
+        value = object[k];
+        return true;
+      }
+      if (object[k] && typeof object[k] === 'object') {
+        value = searchForKey(object[k], key);
+        return value !== undefined;
+      }
+    });
+    return value;
+  }
+  
+  /**
+   * Extract the title of the recipe from the given recipe schema JSON obejct
+   * @param {Object} data Raw recipe JSON to find the image of
+   * @returns {String} If found, returns the recipe title
+   */
+  function getTitle(data) {
+    if (data.name) return data.name;
+    if (data['@graph']) {
+      for (let i = 0; i < data['@graph'].length; i++) {
+        if (data['@graph'][i]['@type'] == 'Recipe') {
+          if (data['@graph'][i]['name']) return data['@graph'][i]['name'];
+        };
+      }
+    }
+    return null;
+  }
+  
+  /**
+   * Extract the yield of the recipe from the given recipe schema JSON obejct
+   * @param {Object} data Raw recipe JSON to find the image of
+   * @returns {String} If found, returns the recipe yield
+   */
+  function getYield(data) {
```

Comment:
Codacy has a fix for the issue: [Expected indentation of 0 spaces but found 2.](https://app.codacy.com/gh/cse110-fa21-group27/cse110-fa21-group27/pullRequest?prid=8447291)
```suggestion
function getYield(data) {
```